### PR TITLE
NOT MERGE Add tabs to Crayons and to the notifications page

### PR DESCRIPTION
**Pull request by ludwiczakpawel**
## What type of PR is this? (check all applicable)

- [x] Feature

## Description

This PR updates styles for Tabs and it adds better responsiveness behaviours. This component has already existed in the codebase in a very basic form and was used in one place: Notifications - i've also updated that.

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)

https://d.pr/i/ND6UIK

## Added tests?

- [x] no, because they aren't needed

## Added to documentation?

- [x] no documentation needed

### CODE REVIEW

File: app/assets/stylesheets/components/forms.scss [hunk 1]
```diff
@@ -113,7 +113,7 @@
 // Selects
 .crayons-select {
   line-height: var(--lh-base);
-  padding: 0.5em;
+  padding: calc(0.5em - var(--border-width)) 0.5em;
```
Comment (ludwiczakpawel) [inline app/assets/stylesheets/components/forms.scss:116]: Just a little bug I found in the meantime....

File: app/views/notifications/index.html.erb [hunk 2]
```diff
@@ -95,95 +95,43 @@
...
-                  <span class="notifications-filter__dropdown__label--subtitle">
-                    @<%= @user.username %> · Personal
-                  </span>
-                </div>
-                <div class="notifications-filter__icon">
-                  <%= inline_svg_tag("checkmark.svg", aria: true, class: "crayons-icon", title: "Checkmark") %>
-                </div>
-              </a>
+          <div class="p-2 s:inline-flex">
+            <select onChange="window.location.href=this.value" class="crayons-select">
```
Comment (rhymes) [inline app/views/notifications/index.html.erb:101]: I'm a bit uneasy about this.

cc. @nickytonline about it

**Comment by ludwiczakpawel** [comment]
> Right now in production we have all this info visible:
> Screenshot 2020-05-08 at 12 36 18 PM
> Instead now we show only the username or slug, could we at least replace it with `user.name` and `organization.name` ?

I think so... But I wanted to be consistent with what we have on desktop version where we display username/slug and not full names. In that case we should probably unify these on desktop version too. Thoughts?

> Also, there's not much padding (between checkbox and name) or margin between the entries in the select
> Screenshot 2020-05-08 at 12 37 00 PM

It's a native browser select - nothing we can control. Current version (the one with full info like name, slug etc) is a bit overkill in terms of information and implementation. After chatting with @lisasy who worked on it earlier, we came to conclusion we can simplify that to be browser-native `select`. If you preview the smallest breakpoint on your desktop browser it looks off - that's true. But you usually open the smallest breakpoint on actual mobile device in which select becomes sliding drawer from the bottom: https://d.pr/v/QQ2OC0

**Comment by rhymes** [comment]
@ludwiczakpawel agreed, let's keep it like this for now. thanks for changing names. I'd like @nickytonline to take a look at this before we merge it but overall, looks awesome!

**Comment by ludwiczakpawel** [comment]
@rhymes cheers! I've added _NOT MERGE_ to the title until Nick review it.

File: app/assets/stylesheets/components/tabs.scss [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+  &__item {
+    padding: var(--tab-vertical-padding) var(--tab-horizontal-padding);
+    display: inline-block;
+    text-align: center;
+    outline: none;
+    font-size: var(--font-size);
+    line-height: var(--su-6);
+    text-decoration: none;
+    cursor: pointer;
```
Comment (nickytonline) [inline app/assets/stylesheets/components/tabs.scss:29]: Just curious about how you make the decision to use a CSS property vs. a utility class? In this case, i.e. `cursor-pointer` utility class vs. a CSS propert/value, `cursor: pointer`.

File: app/views/notifications/index.html.erb (hunk 2 above)
Comment (nickytonline) [reply app/views/notifications/index.html.erb:101]: This should be set up using [addEventListener](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener) via JS. Do you need help with that @ludwiczakpawel? If so, we can sync in the morning/your afternoon.

File: app/assets/stylesheets/components/tabs.scss (hunk 3 above)
Comment (ludwiczakpawel) [reply app/assets/stylesheets/components/tabs.scss:29]: Utility classes are more for scenarios when you want to either customize existing component or for example create w feature quickly without building the component itself. Or you need to build a particular view or piece of view and not sure how to handle that in frontend infra...

I've tried to give few scenarios for utility classes usage in one [comment for Vaidehi](https://github.com/thepracticaldev/dev.to/pull/7732#discussion_r421734732) - I hope that's helpful.

File: app/views/notifications/index.html.erb (hunk 2 above)
Comment (ludwiczakpawel) [reply app/views/notifications/index.html.erb:101]: would you be able to help me out with this one?

not really sure how to implement that

**Pull request merged**